cmake/Dependencies.cmake

```diff
@@ -1324,6 +1324,55 @@ if(USE_ROCM AND ROCM_VERSION_DEV VERSION_LESS "5.2.0")
   include_directories(SYSTEM ${THRUST_PATH})
 endif()
 
+# ---[ XPU
+if(USE_XPU)
+  include(${CMAKE_CURRENT_LIST_DIR}/public/xpu.cmake)
+  # Support GCC only in the moment.
+  if(PYTORCH_FOUND_XPU AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # # -- Host flags (SYCL_CXX_FLAGS)
+    # Combine with torch_compile_options
+    list(APPEND SYCL_HOST_FLAGS -fPIC)
+    list(APPEND SYCL_HOST_FLAGS -std=c++17)
+    # SYCL headers warnings
+    list(APPEND SYCL_HOST_FLAGS -Wno-deprecated-declarations)
+    list(APPEND SYCL_HOST_FLAGS -Wno-attributes)
+
+    if(CMAKE_BUILD_TYPE MATCHES Debug)
+       list(APPEND SYCL_HOST_FLAGS -g)
+       list(APPEND SYCL_HOST_FLAGS -O0)
+    endif(CMAKE_BUILD_TYPE MATCHES Debug)
+
+    # -- Kernel flags (SYCL_KERNEL_OPTIONS)
+    # The fast-math will be enabled by default in SYCL compiler.
+    # Refer to [https://clang.llvm.org/docs/UsersManual.html#cmdoption-fno-fast-math]
+    # 1. We enable below flags here to be warn about NaN and Infinity,
+    # which will be hidden by fast-math by default.
+    # 2. The associative-math in fast-math allows floating point
+    # operations to be reassociated, which will lead to non-deterministic results.
+    # 3. The approx-func allows certain math function calls (such as log, sqrt, pow, etc)
+    # to be replaced with an approximately equivalent set of instructions or
+    # alternative math function calls, which have great errors.
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fno-sycl-unnamed-lambda)
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -sycl-std=2020)
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fhonor-nans)
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fhonor-infinities)
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fno-associative-math)
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -fno-approx-func)
+    # TODO: Align with PyTorch and switch to ABI=0 eventually, after
+    # resolving incompatible implementation in SYCL runtime.
+    set(SYCL_KERNEL_OPTIONS ${SYCL_KERNEL_OPTIONS} -D_GLIBCXX_USE_CXX11_ABI=1)
+    set(SYCL_FLAGS ${SYCL_FLAGS} ${SYCL_KERNEL_OPTIONS})
+
+    if(BUILD_TEST)
+      add_subdirectory(${PROJECT_SOURCE_DIR}/test/cpp/sycl ${CMAKE_BINARY_DIR}/test_sycl)
+    endif()
+  else()
+    message(WARNING "Not compiling with XPU. Could NOT find SYCL."
+      "Suppress this warning with -DUSE_XPU=OFF.")
```

Comment:
Another case is the compiler is not GCC. Suggest to amend the warning message accordingly.